src/System.Net.WebSockets.Client/src/System/Net/WebSockets/WebSocketHandle.Unix.cs

```diff
@@ -2,108 +2,1037 @@
-        }
+            private static byte[] BuildRequestHeader(Uri uri, ClientWebSocketOptions options, string secKey)
+            {
+                StringBuilder builder = t_cachedStringBuilder ?? (t_cachedStringBuilder = new StringBuilder());
+                Debug.Assert(builder.Length == 0, $"Expected builder to be empty, got one of length {builder.Length}");
+                try
+                {
+                    builder.Append("GET ").Append(uri.PathAndQuery).Append(" HTTP/1.1\r\n");
 
-        public Task CloseAsync(
-            WebSocketCloseStatus closeStatus,
-            string statusDescription,
-            CancellationToken cancellationToken)
-        {
-            throw new PlatformNotSupportedException(SR.net_WebSockets_UnsupportedPlatform);
-        }
+                    // Add all of the required headers
+                    builder.Append("Host: ").Append(uri.Host).Append("\r\n");
+                    builder.Append("Connection: Upgrade\r\n");
+                    builder.Append("Upgrade: websocket\r\n");
+                    builder.Append("Sec-WebSocket-Version: 13\r\n");
+                    builder.Append("Sec-WebSocket-Key: ").Append(secKey).Append("\r\n");
 
-        public Task CloseOutputAsync(
-            WebSocketCloseStatus closeStatus,
-            string statusDescription,
-            CancellationToken cancellationToken)
-        {
-            throw new PlatformNotSupportedException(SR.net_WebSockets_UnsupportedPlatform);
-        }
+                    // Add all of the additionally requested headers
+                    foreach (string key in options.RequestHeaders.AllKeys)
+                    {
+                        builder.Append(key).Append(": ").Append(options.RequestHeaders[key]).Append("\r\n");
+                    }
 
-        public void Dispose()
-        {
-        }
+                    // Add the optional subprotocols header
+                    if (options.RequestedSubProtocols.Count > 0)
+                    {
+                        builder.Append(HttpKnownHeaderNames.SecWebSocketProtocol).Append(": ");
+                        builder.Append(options.RequestedSubProtocols[0]);
+                        for (int i = 1; i < options.RequestedSubProtocols.Count; i++)
+                        {
+                            builder.Append(", ").Append(options.RequestedSubProtocols[i]);
+                        }
+                        builder.Append("\r\n");
+                    }
 
-        public void Abort()
-        {
+                    // Add an optional cookies header
+                    if (options.Cookies != null)
+                    {
+                        string header = options.Cookies.GetCookieHeader(uri);
+                        if (!string.IsNullOrWhiteSpace(header))
+                        {
+                            builder.Append(HttpKnownHeaderNames.SetCookie).Append(": ").Append(header).Append("\r\n");
```

Comment:
Do we want to do proactive (or even reactive) header chopping for HTTP 413 protection?  (A quick search says the default for Apache is/was 8k per header, and IIS is/was 16k; but those are/were defaults).

So something like  if (header.Length > 8100) { find a semicolon before that mark, send up to there, newline; repeat ad infinitum }